pandas/tests/util/test_assert_frame_equal.py

```diff
@@ -323,9 +323,22 @@ def test_frame_equal_mixed_dtypes(frame_or_series, any_numeric_ea_dtype, indexer
         tm.assert_equal(obj1, obj2, check_exact=True, check_dtype=False)
 
 
-def test_assert_series_equal_check_like_different_indexes():
+def test_assert_frame_equal_check_like_different_indexes():
     # GH#39739
     df1 = DataFrame(index=pd.Index([], dtype="object"))
     df2 = DataFrame(index=pd.RangeIndex(start=0, stop=0, step=1))
     with pytest.raises(AssertionError, match="DataFrame.index are different"):
         tm.assert_frame_equal(df1, df2, check_like=True)
+
+
+def test_assert_frame_equal_checking_allow_dups_flag():
+    # GH#45554
+    left = DataFrame([[1, 2], [3, 4]])
+    left.flags.allows_duplicate_labels = False
+
+    right = DataFrame([[1, 2], [3, 4]])
+    right.flags.allows_duplicate_labels = True
```

Comment:
do we also need to test for DataFrames with the same flags with check_flags=True/False. I find it odd (without investigation) that the reported issue was not picked up before. The flag is on the DataFrame so why not check that instead of on the Series?